mono/mini/debugger-agent.c

```diff
@@ -5204,6 +5205,12 @@ static void
 debugger_agent_handle_exception (MonoException *exc, MonoContext *throw_ctx,
 									  MonoContext *catch_ctx, StackFrameInfo *catch_frame)
 {
+	if (catch_ctx == NULL && catch_frame == NULL && mini_debug_options.suspend_on_unhandled && mono_object_class (exc) != mono_defaults.threadabortexception_class) {
+		mono_runtime_printf_err ("Unhandled exception, suspending...");
+		while (1)
+			;
+	}
```

Comment:
If the loop is here, doesn't that mean that `suspend-on-unhandled` will only be triggered when we start with debugging?   I think `suspend-on-unhandled` is supposed to work even without the debugger enabled.

Reply:
I don't think so. I've tested using this command line:
- MONO_DEBUG=suspend-on-unhandled ./mono/mini/mono-sgen test16588.exe
And suspend-on-unhandled was triggered.